spec/mongoid/document_persistence_context_spec.rb

```diff
@@ -63,6 +63,18 @@
         person.update username: 'zyg15'
         expect(Person.with(options) { Person.first.username }).to be == 'zyg15'
       end
+
+      it 'an explicit context takes precedence over a remembered context when persisting' do
+        person.username = 'bob'
+        # should not actually save -- the person does not exist in the
+        # `other` collection and so cannot be updated.
```

Comment:
Doesn't this do an upsert, so it will create it as a new record in the 'other' collection?

Reply:
Hmm... that is surprising to me. Calling `.save!` and *silently not persisting* does not seem like acceptable behavior to me. As a user I'd prefer it either to upsert, or if that's not possible, have `.save!` throw an error and `.save` return false.

Reply:
I agree, it's not great behavior. I opened https://jira.mongodb.org/browse/MONGOID-5712 last week to investigate this more.